src/Service/PromotionsService.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Service;
+
+use BBC\ProgrammesPagesService\Cache\CacheInterface;
+use BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository\PromotionRepository;
+use BBC\ProgrammesPagesService\Domain\Entity\Programme;
+use BBC\ProgrammesPagesService\Domain\Exception\DataNotFetchedException;
+use BBC\ProgrammesPagesService\Mapper\ProgrammesDbToDomain\PromotionMapper;
+use DateTimeImmutable;
+
+class PromotionsService extends AbstractService
+{
+    /** @var PromotionRepository */
+    protected $repository;
+
+    public function __construct(
+        PromotionRepository $repository,
+        PromotionMapper $mapper,
+        CacheInterface $cache
+    ) {
+        parent::__construct($repository, $mapper, $cache);
+    }
+
+    /**
+     * @return Promotion[]
+     * @throws DataNotFetchedException if we get a promotion but there is nothing being
+     *     promoted (an image or core entity) associated to the promotion
+     */
+    public function findActivePromotionsByProgramme(
+        Programme $programme,
+        DateTimeImmutable $dateTime,
```

Comment:
The convention is that repository methods require no external state (such as a global application time that is not provided as part of the params). This allows them to be flexible so that we have the potential to reuse queries across multiple services methods. They are the core of our library so they should be functional (in the only use parameters, no side-effects sense).

Services meanwhile are a bit more fixed with their functionality, "getActivePromotions" is a common piece of functionality and we're never going to need to find promotions that were active at a point other than now. Services are the point where we encapsulate business logic, so they can have a bit more global state.

Thus I think this is the right layer to make use of ApplicationTime, rather than within the Controller (a layer up) or within the Repo (a later down).